src/others/font_manager/lv_font_manager_backends.c

```diff
@@ -0,0 +1,221 @@
+/**
+ * @file lv_font_manager_backends.c
+ *
+ */
+
+/*********************
+ *      INCLUDES
+ *********************/
+
+#include "lv_font_manager_backends.h"
+
+#if LV_USE_FONT_MANAGER
+
+#include "../../libs/freetype/lv_freetype.h"
+#include "../../libs/tiny_ttf/lv_tiny_ttf.h"
+#include "../../stdlib/lv_sprintf.h"
+
+/*********************
+ *      DEFINES
+ *********************/
+
+/**********************
+ *      TYPEDEFS
+ **********************/
+
+/**********************
+ *  STATIC PROTOTYPES
+ **********************/
+
+static lv_font_t * builtin_font_create_cb(const lv_font_info_t * info, const void * src);
+static void builtin_font_delete_cb(lv_font_t * font);
+static void * builtin_font_dup_src_cb(const void * src);
+static void builtin_font_free_src_cb(void * src);
+
+#if LV_USE_FREETYPE
+    static lv_font_t * freetype_font_create_cb(const lv_font_info_t * info, const void * src);
+    static void freetype_font_delete_cb(lv_font_t * font);
+    static void * freetype_font_dup_src_cb(const void * src);
+    static void freetype_font_free_src_cb(void * src);
+#endif /*LV_USE_FREETYPE*/
+
+#if LV_USE_TINY_TTF
+    static lv_font_t * tiny_ttf_font_create_cb(const lv_font_info_t * info, const void * src);
+    static void tiny_ttf_font_delete_cb(lv_font_t * font);
+    static void * tiny_ttf_font_dup_src_cb(const void * src);
+    static void tiny_ttf_font_free_src_cb(void * src);
+#endif /*LV_USE_TINY_TTF*/
+
+/**********************
+ *  STATIC VARIABLES
+ **********************/
+
+const lv_font_class_t lv_builtin_font_class = {
+    .create_cb = builtin_font_create_cb,
+    .delete_cb = builtin_font_delete_cb,
+    .dup_src_cb = builtin_font_dup_src_cb,
+    .free_src_cb = builtin_font_free_src_cb,
+};
+
+#if LV_USE_FREETYPE
+const lv_font_class_t lv_freetype_font_class = {
+    .create_cb = freetype_font_create_cb,
+    .delete_cb = freetype_font_delete_cb,
+    .dup_src_cb = freetype_font_dup_src_cb,
+    .free_src_cb = freetype_font_free_src_cb,
+};
+#endif /*LV_USE_FREETYPE*/
+
+#if LV_USE_TINY_TTF
+const lv_font_class_t lv_tiny_ttf_font_class = {
+    .create_cb = tiny_ttf_font_create_cb,
+    .delete_cb = tiny_ttf_font_delete_cb,
+    .dup_src_cb = tiny_ttf_font_dup_src_cb,
+    .free_src_cb = tiny_ttf_font_free_src_cb,
+};
+#endif /*LV_USE_TINY_TTF*/
```

Comment:
As these are generic font concepts, they should be moved to their own libraries for better scalability and resuability.